libraries/apollo-ast/src/commonMain/kotlin/com/apollographql/apollo3/ast/internal/semanticEquals.kt

```diff
@@ -1,183 +1,200 @@
 package com.apollographql.apollo3.ast.internal
 
+import com.apollographql.apollo3.ast.GQLArgument
 import com.apollographql.apollo3.ast.GQLBooleanValue
 import com.apollographql.apollo3.ast.GQLDirective
 import com.apollographql.apollo3.ast.GQLDirectiveDefinition
 import com.apollographql.apollo3.ast.GQLEnumTypeDefinition
 import com.apollographql.apollo3.ast.GQLEnumValue
 import com.apollographql.apollo3.ast.GQLFloatValue
+import com.apollographql.apollo3.ast.GQLInputValueDefinition
 import com.apollographql.apollo3.ast.GQLIntValue
 import com.apollographql.apollo3.ast.GQLListType
 import com.apollographql.apollo3.ast.GQLListValue
+import com.apollographql.apollo3.ast.GQLNamed
 import com.apollographql.apollo3.ast.GQLNamedType
+import com.apollographql.apollo3.ast.GQLNode
 import com.apollographql.apollo3.ast.GQLNonNullType
 import com.apollographql.apollo3.ast.GQLNullValue
 import com.apollographql.apollo3.ast.GQLObjectValue
 import com.apollographql.apollo3.ast.GQLStringValue
-import com.apollographql.apollo3.ast.GQLType
-import com.apollographql.apollo3.ast.GQLValue
 import com.apollographql.apollo3.ast.GQLVariableValue
 import com.apollographql.apollo3.ast.toUtf8
 
-internal fun GQLDirectiveDefinition.semanticEquals(other: GQLDirectiveDefinition): Boolean {
-  if (name != other.name) {
-    return false
-  }
-
-  if (locations != other.locations) {
-    return false
-  }
-
-  if (repeatable != other.repeatable) {
-    return false
-  }
+internal fun GQLNode.semanticEquals(other: GQLNode): Boolean {
+  when (this) {
+    is GQLDirectiveDefinition -> {
+      if (other !is GQLDirectiveDefinition) {
+        return false
```

Comment:
It's handled later with `if (this is GQLNamed) {`

Reply:
I see 👍

Reply:
It's currently a bit weird though because the check is done twice for `GQLVariableValue` for an example. It implements `GQLNamed` but also makes the explicit check